packages/kbn-esql-utils/src/utils/get_esql_adhoc_dataview.ts

```diff
@@ -31,11 +31,19 @@ export async function getESQLAdHocDataview(
   indexPattern: string,
   dataViewsService: DataViewsPublicPluginStart
```

Comment:
I wonder if it makes sense to provide a flag for this function consumers to disable the automatic `@timestamp` injection.

Something like:

```ts
export async function getESQLAdHocDataview(
  indexPattern: string,
  dataViewsService: DataViewsPublicPluginStart,
  options: { injectTimeField: boolean } = { injectTimeField: true } 
) {
```

Reply:
No Marco, I dont want this now. I want all the consumers of this utility to work like this otherwise we are losing the consistent behavior across kibana. Is on purpose like that

Reply:
I proposed that as I see some of the replaced snippets having slightly different criteria variants on whether to add the `timeField` or not.
Does it mean that all those were buggy?

Reply:
yes they we re buggy, exactly